src/btf.cpp

```diff
@@ -456,16 +456,19 @@ void BTF::resolve_args(const std::string &func,
     SizedType stype = get_stype(BTFId{ .btf = func_id.btf, .id = p->type });
     stype.funcarg_idx = j;
     stype.is_funcarg = true;
-    args.insert({ str, stype });
+    args.AddField(str, stype, args.size, std::nullopt, false);
+    args.size += stype.GetSize();
```

Comment:
Do we need to consider padding inside the struct?

Reply:
Good catch. In fact, kprobe (fentry/fexit) arguments are stored in a u64 array, so we always need to add 8 here. It's not a problem that we represent them using a struct b/c we use `funcarg_idx` to access them in codegen, but I added a comment explaining that to prevent confusion in future.